README.md

```diff
@@ -421,6 +421,26 @@ The following settings are optional and allow you to further configure your clus
 * `settings.kubernetes.container-log-max-size`: The maximum size of container log file before it is rotated.
 * `settings.kubernetes.cpu-manager-policy`: Specifies the CPU manager policy. Possible values are `static` and `none`. Defaults to `none`. If you want to allow pods with certain resource characteristics to be granted increased CPU affinity and exclusivity on the node, you can set this setting to `static`. You should reboot if you change this setting after startup - try `apiclient reboot`.
 * `settings.kubernetes.cpu-manager-reconcile-period`: Specifies the CPU manager reconcile period, which controls how often updated CPU assignments are written to cgroupfs. The value is a duration like `30s` for 30 seconds or `1h5m` for 1 hour and 5 minutes.
+* `settings.kubernetes.credential-providers`: Contains a collection of Kubelet image credential provider settings.
+
+  Example user data for setting up image credential providers:
+
+  ```toml
+  [[settings.kubernetes.credential-providers]]
+  enabled = true
+  name = "ecr-credential-provider"
+  # (optional - defaults to "12h")
+  cache-duration = "30m"
+  image-patterns = [
+    # One or more URL paths to match an image prefix. Supports globbing of subdomains.
+    "*.dkr.ecr.us-east-2.amazonaws.com",
+    "*.dkr.ecr.us-west-2.amazonaws.com"
+  ]
+  ```
```

Comment:
Elsewhere we've preferred maps rather than lists to represent these "named" structures, e.g. `settings.host-containers` and `settings.pki`.

One of the main reasons is that it's more user-friendly to do targeted edits to a specific entry:
```
apiclient set kubernetes.credential-providers.ecr-credential-provider.enabled=true
```

With a list, I believe the entire list would need to be provided again, in the JSON syntax.